### ISSUE TITLE
Using alignment and uniformity as metrics only.

### ISSUE THREAD

**Issue by NaderEK** [SsnL/align_uniform#10]
Dear Tongzhou,

Thanks for the great work - I am wondering if I can use alignment and uniformity losses as metrics only to quantify the alignment and uniformity of the features of other SSL encoders like MAE without training a model from scratch using these losses.

I get -inf values for uniform loss when the batch size is small. However, when I increase it to 10 onwards, it outputs normal -ve values.
I also get ranges of [5,20] for alignment and [-50,-200] uniformity losses when I run the code below.

```
for x,y in loader:
    x, y = encoder.features(torch.cat([x.to('cuda'), y.to('cuda')])).chunk(2)
    x,y = x.detach(), y.detach()
    x = x[:, 1:].mean(dim=1)
    y = y[:, 1:].mean(dim=1)

    align_loss_val = align_loss(x, y, alpha=2)
    unif_loss_val = (uniform_loss(x, t) + uniform_loss(y, t)) / 2
    loss = align_loss_val * 1 + unif_loss_val * 1
    align_meter.update(align_loss_val, x.shape[0])
    unif_meter.update(unif_loss_val)
    loss_meter.update(loss, x.shape[0])
```

Is there something I should consider to quantify these properties in the features coming out of the encoder?

Thank you

**Comment by ssnl** [comment]
Hi @SirNader , thanks for the question! Alignment and uniform metrics are defined for features that are l2-normalized, which is what contrastive encoders generally outut. For non-contrastive encoders, they usually don't output normalized features, but you can definitely normalize them and then evaluate. These metrics will still give you a sense of the output distribution, but I also don't know how well they correlate with representation quality (in terms of downstream tasks) for non-contrastive encoders. It would be cool to find out, for sure!

**Comment by NaderEK** [comment]
Thank you, @ssnl! that was helpful
